packages/java-edition/src/mcfunction/index.ts

```diff
@@ -70,6 +71,84 @@ export const initialize = (
 		),
 	)
 
+	mcdoc.runtime.registerAttribute(
+		meta,
+		'command_argument',
+		mcdoc.runtime.attribute.validator.map(
+			mcdoc.runtime.attribute.validator.string,
+			core.ResourceLocation.lengthen,
+		),
+		{
+			attachString: (config, ctx) => {
+				let argParser
+				try {
+					argParser = parser.argument({
+						type: 'argument',
+						parser: config,
+					})
+				} catch (e) {
+					ctx.logger.warn(
+						`[mcdoc command_argument] Failed to create parser ${config}`,
+					)
+					return undefined
+				}
+				return (node) => {
+					const src = new core.Source(node.value, node.valueMap)
+					if (!argParser) {
+						ctx.err.report(
+							localize(
+								'mcfunction.parser.unknown-parser',
+								localeQuote(config),
+							),
+							core.Range.create(src.cursor, src.skipRemaining()),
+							core.ErrorSeverity.Hint,
+						)
+						return
+					}
+					const child = argParser(src, ctx)
+					if (child === core.Failure) {
+						ctx.err.report(
+							localize('mcfunction.parser.eoc-unexpected'),
+							node,
+						)
+						return
+					}
+					if (src.canRead()) {
+						ctx.err.report(
+							localize('mcdoc.runtime.checker.trailing'),
+							core.Range.create(src.cursor, src.skipRemaining()),
+						)
+					}
+					node.children = [child]
```

Comment:
Can this cause issues / overwrite sth? Would it be better to do this?
```ts
node.children ??= []
node.children.push(child)
```

Reply:
I refactored this later where the logic is now in only one place in the checker, and it only runs once per node during the `attachTypeInfo`. We don't need to support the use case where multiple string parsers run on the same node, that's never going to work.